src/cbrkit/sim/collections.py

```diff
@@ -90,11 +90,15 @@ def __call__(self, x: Sequence[V], y: Sequence[V]) -> float:
     pass
 
 try:
+    from dataclasses import dataclass
+    from typing import Collection, Callable, Any, List, Tuple, Union
```

Comment:
These should be added as top-level imports as they are always available. Also note that ruff reports some errors due to deprecated imports from typing, these must be resolved or the CI checks will fail. They must either be imported from `collections.abc` or use their generic alternatives directly: https://typing.readthedocs.io/en/latest/guides/modernizing.html